compiler/rustc_hir_analysis/src/check/method/suggest.rs

```diff
@@ -1201,6 +1203,68 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    fn suggest_range_for_iter(
+        &self,
+        tcx: TyCtxt<'tcx>,
+        source: SelfSource<'_>,
+        span: Span,
+        item_name: Ident,
+        ty_str: &str,
+    ) -> bool {
+        if let SelfSource::MethodCall(expr) = source {
+            // avoid suggesting when the method name is not implemented for a `range`
+            // FIXME: any better solution?
+            let valid_methods = [
```

Comment:
Any better solution for this? @lcnr

Reply:
yeah, that's obviously not great :thinking:

I guess what you can do is:

- first check if the integer is the second half of a range
- probe whether this method is found for ranges

lint in that case

I am not 100% sure how to best do that: probably just using `lookup_probe` :thinking: haven't used the method code too much